flask_menu/__init__.py

```diff
@@ -308,6 +342,110 @@ def _register_menu_item():
     return menu_decorator
 
 
+def register_flaskview(app, classy_view):
+    """Register a Flask-Classy FlaskView's menu items with the menu register.
+
+    Example::
+
+        bp = Blueprint('bp', __name__)
+        menu.register_flaskview(bp, MyEndpoint)
+
+    :param app: Application or Blueprint which owns the
+        function view.
+    :param classy_view: The Flask-Classy FlaskView class to register
+        menu items for.
+    """
+    if isinstance(app, Blueprint):
+        endpoint_prefix = app.name + '.'
+        before_first_request = app.before_app_first_request
+    else:
+        endpoint_prefix = ''
+        before_first_request = app.before_first_request
+
+    @before_first_request
+    def _register_menu_items():
+        for meth_str in dir(classy_view):
+            meth = getattr(classy_view, meth_str)
+
+            if hasattr(meth, '_menu_items'):
+                for menu_item in meth._menu_items:
+                    endpoint = "{}{}:{}".format(
+                        endpoint_prefix,
+                        classy_view.__name__,
+                        meth.__name__
+                    )
+                    path = menu_item.pop('path')
+                    item = current_menu.submenu(path)
+                    item.register(
+                        endpoint,
+                        **menu_item
+                    )
+
+
+def classy_menu_item(path, text, **kwargs):
```

Comment:
It would be nice to find a more generic name, however the URL endpoints that classy generates (and this implementation relies on) probably aren't too portable. Classy generates it's endpoints as 'Classname:methodname', see https://github.com/inveniosoftware/flask-menu/pull/33/files#diff-3839e0a4ca06ff71b86a56b51eafdf11R373.